signing_service/signing_service/signing_service.py

```diff
@@ -126,30 +144,72 @@ def _connect(self, tcp_socket: socket.socket) -> None:
-                middleman_message_response = ErrorFrame(
-                    payload=(ErrorCode.InvalidFrameSignature, str(exception)),
-                    request_id=REQUEST_ID_FOR_RESPONSE_FOR_INVALID_FRAME,
+                middleman_message = AbstractFrame.deserialize(
+                    raw_message_received,
+                    public_key=self.concent_public_key,
                 )
-                logger.debug(
-                    f'Sending Middleman protocol message with request_id: {middleman_message_response.request_id}.'
-                )
-                send_over_stream(
-                    connection=tcp_socket,
-                    raw_message=middleman_message_response,
+
+                if (
+                    not middleman_message.payload_type == PayloadType.GOLEM_MESSAGE or
+                    not isinstance(middleman_message.payload, TransactionSigningRequest)
+                ):
+                    raise SigningServiceUnexpectedMessageError
+
+            # Is the frame correct according to the protocol? If not, error code is InvalidFrame.
+            except (
+                FrameInvalidMiddlemanProtocolError,
+                PayloadTypeInvalidMiddlemanProtocolError,
+                RequestIdInvalidTypeMiddlemanProtocolError,
+            ) as exception:
+                middleman_message_response = self._prepare_error_response(ErrorCode.InvalidFrame, exception)
+            # Is frame signature correct? If not, error code is InvalidFrameSignature.
+            except SignatureInvalidMiddlemanProtocolError as exception:
+                middleman_message_response = self._prepare_error_response(ErrorCode.InvalidFrameSignature, exception)
+            # Is the content of the message valid? Do types match the schema and all values are within allowed ranges?
+            # If not, error code is InvalidPayload.
+            # Can the payload be decoded as a Golem message? If not, error code is InvalidPayload.
+            # Is payload message signature correct? If not, error code is InvalidPayload.
+            except (MessageError, PayloadInvalidMiddlemanProtocolError) as exception:
+                middleman_message_response = self._prepare_error_response(ErrorCode.InvalidPayload, exception)
+            # Is frame type GOLEM_MESSAGE? If not, error code is UnexpectedMessage.
+            # Is Golem message type TransactionSigningRequest? If not, error code is UnexpectedMessage.
+            except SigningServiceUnexpectedMessageError as exception:
+                middleman_message_response = self._prepare_error_response(ErrorCode.UnexpectedMessage, exception)
+            # If received frame is correct, validate transaction.
+            else:
+                golem_message_response = self._get_signed_transaction(middleman_message.payload)
+                golem_message_response.sign_message(
                     private_key=self.signing_service_private_key,
                 )
```

Comment:
If you have one line please don't brake this to separate lines.

Reply:
Ok, I have always found breaking more readable, but makes sens for single, not-so-long arguments.

Reply:
done :+1: